client/src/main/scala/org/http4s/client/middleware/GZip.scala

```diff
@@ -0,0 +1,47 @@
+package org.http4s
+package client
+package middleware
+
+import cats.effect.Sync
+import org.http4s.headers.{`Accept-Encoding`, `Content-Encoding`}
+
+/**
+  * Client middleware for enabling gzip.
+  */
+object GZip {
+
+  private val supportedCompressions =
+    Seq(ContentCoding.gzip.coding, ContentCoding.deflate.coding).mkString(", ")
+
+  def apply[F[_]: Sync](bufferSize: Int = 32 * 1024)(client: Client[F]): Client[F] =
```

Comment:
Is a Sync constraint necessary?

Reply:
fs2.compress.gunip requires an F that is RaiseThrowable. Since RaiseThrowable is from fs2, I added Sync here. Can I use any other type instead?

Reply:
`ApplicativeError`?

Reply:
I tried changing it to ApplicativeError. Now, I'm getting missing Bracket error for new Client creation. I'll update PR by changing this to Bracket

Reply:
Updated PR